Modules/_datetimemodule.c

```diff
@@ -2896,6 +2900,36 @@ date_fromordinal(PyObject *cls, PyObject *args)
     return result;
 }
 
+
+/* Return new date from time.strptime(). */
+static PyObject *
+date_strptime(PyObject *cls, PyObject *args)
+{
+    PyObject *date = NULL;
+    PyObject *datetime;
+
+    datetime = datetime_strptime((PyObject *)&PyDateTime_DateTimeType, args);
+
+    if (datetime == NULL) {
+        return NULL;
+    }
+
+    if (DATE_GET_HOUR(datetime) ||
```

Comment:
Hm, if I understand correctly, this is inconsistent with the pure Python version. The pure Python version checks if there are any time components in the *format string*, whereas the C version parses to a datetime and then checks to see if there are any *time components*.

I think checking the format string is the better way to do this, as I mention in another comment, this approach seems to indicate that `date.strptime("2018-01-01 00:00", "%Y-%m-%d %H:%M")` would *succeed*, which is not the right thing to do. I will comment on the test suite to add a test for this.

Reply:
ok, I will check the Python implementation, but in this case, we could migrate the Python implementation to the C layer?

Reply:
Yeah, port the Python implementation to the C layer.

Reply:
lol ;-) I am not an expert with the C-API, but I could try, it's a good exercise for my comprehension of the C-API of Python

Reply:
Yeah, you have to reimplement _check_invalid_datetime_specs() in C.

Reply:
Why does it have to be implemented in C? There are already calls out to the __strptime_ Python-language module here.

Reply:
not sure, but because there is an other test in C for the date, and maybe we could put the verification process in only one function. @vstinner do you confirm?